src/components/pages/course-page/CoursePage.tsx

```diff
@@ -0,0 +1,135 @@
+import React, { FC, useMemo, useState } from 'react';
+
+import { Box, Divider, List, Rating, Typography } from '@mui/material';
+import { useQuery } from 'react-query';
+import CourseService from '@/services/course.service';
+import Loader from '@/components/common/loader';
+import { useRouter } from 'next/router';
+import {
+  AccessTime,
+  CalendarToday,
+  CastForEducation,
+  Check,
+} from '@mui/icons-material';
+import Lesson from '@/components/pages/course-page/components/lesson/Lesson';
+import { parseCourseCover, parseDate, parseTime, sortLessons } from '@/utils';
+
+import styles from './CoursePage.module.scss';
+import Tag from '@/components/common/tag';
+import IconField from '@/components/pages/course-page/components/icon-field';
+import { showToast } from '@/redux/reducers/toast.reducer';
+import { TOAST_STATUS } from '@/types/redux/toast';
+import { useDispatch } from 'react-redux';
+
+const CoursesPage: FC = () => {
+  const { query, push } = useRouter();
+  const courseId = query.courseId as string;
+
+  const { data: course, isLoading } = useQuery(
+    ['courses', courseId],
+    () => CourseService.getCourse(courseId),
+    { refetchOnWindowFocus: false, retry: false },
+  );
+
+  const lessonText = useMemo(() => {
+    const lessonsNum = course?.lessons.length;
+    const closedLessonsNum = course?.lessons.filter(
```

Comment:
More understandable name would be `lockedLessonsNum`

Reply:
fixed